StortSpelProjekt/Engine/src/AudioEngine/AudioVoice.cpp

```diff
@@ -79,3 +82,37 @@ void AudioVoice::Stop()
         m_pSourceVoice->SubmitSourceBuffer(m_pBuffer);
     }
 }
+
+void AudioVoice::Resume()
+{
+    if (m_Initialized)
+    {
+        // reset the buffer if sound has ended
+        XAUDIO2_VOICE_STATE test;
+        m_pSourceVoice->GetState(&test);
+        if (test.BuffersQueued == 0)
+        {
+            // reset the buffer so the sound starts from the beginning at next playback
+            m_pSourceVoice->FlushSourceBuffers();
+            m_pSourceVoice->SubmitSourceBuffer(m_pBuffer);
+        }
+
+        // play the sound
+        if (FAILED(m_pSourceVoice->Start(0)))
+        {
+            Log::Print("Error playing audio\n");
+        }
+    }
+}
+
+void AudioVoice::Pause()
+{
+    if (m_Initialized)
+    {
+        // stop playback
+        if (FAILED(m_pSourceVoice->Stop(0)))
+        {
+            Log::Print("Error stopping audio\n");
+        }
```

Comment:
If this is a print to warn the developer that the sound has not successfully stopped playing, please make it a warning insted.